### PR TITLE
retroarch: 1.8.1 -> 1.8.5

### PR DESCRIPTION
**Pull request by kolbycrouch**
###### Motivation for this change

Package is out of date, so are the optional core libraries.

###### Things done

Updated retroarch to 1.8.4
Updated all libretro cores.
Fixed broken libretro cores.
Fixed building on aarch64-linux ( Assumes desktop OpenGL ).
Added several missing libretro cores.

All cores should now build on x86_64-linux and aarch64-linux.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kolbycrouch** [comment]
I don't understand the check failure here.

File: pkgs/misc/emulators/retroarch/cores.nix [hunk 1]
```diff
@@ -54,46 +56,33 @@ in with stdenv.lib.licenses;
 
 {
 
-  _4do = (mkLibRetroCore rec {
```
Comment (matthewbauer) [inline pkgs/misc/emulators/retroarch/cores.nix]: Doesn't this one still exist?

File: pkgs/misc/emulators/retroarch/default.nix [hunk 2]
```diff
@@ -50,7 +50,7 @@ in stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  configureFlags = if stdenv.isLinux then [ "--enable-kms" ] else "";
+  configureFlags = stdenv.lib.optionals stdenv.isLinux [ "--enable-kms" "--enable-egl" ];
```
Comment (matthewbauer) [inline pkgs/misc/emulators/retroarch/default.nix]: Can you confirm that "egl" isn't detected by default? I thought in the past the configure script could guess it.

Comment (kolbycrouch) [reply pkgs/misc/emulators/retroarch/default.nix]: I thought that it didn't pick it up when I tested it on aarch64. Maybe it does though. I'm still afraid to change anything until I understand why the check is failing, if it matters.

**Review by matthewbauer — APPROVED** [review]
Looks good.

I have a few changes related to retroarch cores.nix that I have pushed. Mostly have nothing to do with your changes but make things a little less crazy with override logic.

**Comment by matthewbauer** [comment]
I'll merge given this passes CI.

**Comment by matthewbauer** [comment]
@grahamcofborg build libretro

**Comment by matthewbauer** [comment]
@GrahamcOfBorg eval

**Comment by kolbycrouch** [comment]
Thanks for fixing this up for me. I still don't understand the check failure from before. Might it have something to do with setting gw license to just zlib? Other than that I'm clueless.

**Comment by matthewbauer** [comment]
> Might it have something to do with setting gw license to just zlib?

Yeah - the license was being set to zlib the package, not zlib the license. Needed to add stdenv.lib.licenses to make it clear.

**Pull request merged**